src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -65178,5 +65292,162 @@ static void F3(int? z, int? w)
             var comp = CreateCompilationWithIndexAndRange(source, options: WithNonNullTypesTrue());
             comp.VerifyDiagnostics();
         }
+
+        [Fact]
+        public void IsDeclarationPattern_01()
+        {
+            var source =
+@"class Program
+{
+    static void F1(object x1)
+    {
+        if (x1 is string y1)
+        {
+            x1.ToString();
+            y1.ToString();
+        }
+        x1.ToString();
+    }
+    static void F2(object? x2)
+    {
+        if (x2 is string y2)
+        {
+            x2.ToString();
+            y2.ToString();
```

Comment:
>y2 [](start = 12, length = 2)

Is [https://github.com/dotnet/roslyn/issues/30952](https://github.com/dotnet/roslyn/issues/30952) got fixed, or are we not getting any warnings here because the flow state is oblivious? Consider adding  ```y2?.ToString();``` above this line (and other similar lines) to verify that we will get a diagnostics about an unnecessary null check. Also, all changes related to Declaration Pattern are not related to stated subject of this PR, consider separating them into another PR in order to avoid unnecessary friction with this PR.